conversions/src/Data/Conversions.hs

```diff
@@ -141,6 +153,18 @@ instance (MonadError (BoundError Integer Word16) m) => Conversion (m Word16) Int
 instance (MonadError (BoundError Integer Word8) m) => Conversion (m Word8) Integer where
   convert = convertBoundedFromIntegral
 
+instance (MonadError (BoundError Int Word64) m) => Conversion (m Word64) Int where
+  convert = checkedFromIntegralToBounded
+
+instance (MonadError (BoundError Int Word32) m) => Conversion (m Word32) Int where
+  convert = checkedFromIntegralToBounded
+
+instance (MonadError (BoundError Int Word16) m) => Conversion (m Word16) Int where
+  convert = convertBoundedFromIntegral
+
+instance (MonadError (BoundError Int Word8) m) => Conversion (m Word8) Int where
+  convert = convertBoundedFromIntegral
```

Comment:
__question:__ I'm curious about the ordering? I would've expected `Word8`, `Word16`, `Word32`, `Word64` in that order (from smallest to largest). Just wondering if you are using a different scheme.

Reply:
The existing code for Integer to Word conversions is from largest to smallest, so I went with it for the int to word conversions. As far as I remember I don't think there is any deep reason for this scheme.

Reply:
@dkubb Would you consider the ordering blocking, if not, let's get this PR merged?